src/lib.rs

```diff
@@ -28,6 +24,16 @@ pub enum LeopardError {
     #[error("Maximum shard number ({}) exceeded: {0}", ORDER)]
     MaxShardNumberExceeded(usize),
 
+    /// Maximum number of parity shards exceeded.
+    #[error("Maximum parity shard number ({0}) exceeded: {1}")]
+    MaxParityShardNumberExceeded(usize, usize),
+
+    /// This amount of (data, parity) shards is not supported by Leopard algorithm
+    /// and would result in buffer overflow on skew lookup table during encoding.
+    /// Please try using different amounts of shards.
+    #[error("Unsupported amount of data ({0}) and parity ({1}) shards")]
```

Comment:
```suggestion
    #[error("Unsupported number of data ({0}) and parity ({1}) shards")]
```
"amount" usually refers to uncountable nouns, while "number" to countable.